x/sisu/types/tx_out.go

```diff
@@ -1,17 +1,26 @@
 package types
 
 import (
+	"encoding/hex"
 	"fmt"
+	"strings"
 
 	"github.com/sisu-network/lib/log"
+	"golang.org/x/crypto/sha3"
 )
 
 // GetId returns a unique id of the txOut. This is the hash of tx out type, its content and list
 // of transfer id inputs.
 func (t *TxOut) GetId() string {
 	switch t.TxType {
-	case TxOutType_TRANSFER_OUT:
-		return fmt.Sprintf("%s__%s", t.Content.OutChain, t.Content.OutHash)
+	case TxOutType_TRANSFER:
+		hash := sha3.NewLegacyKeccak256()
+		hash.Write([]byte(strings.Join(t.Input.TransferUniqIds, "")))
```

Comment:
The TxOut is also used for non-transfer Id (such as command to change admin, withdraw fund of the smart contract. We should think about that case here.

Reply:
This might be ok for admin command as we will have to make sure all the command ids are unique.